pxr/imaging/plugin/hioImageIO/imageIOImage.mm

```diff
@@ -0,0 +1,891 @@
+        case HioFormatUNorm8Vec4:
+        case HioFormatUNorm8srgb:
+        case HioFormatUNorm8Vec2srgb:
+        case HioFormatUNorm8Vec3srgb:
+        case HioFormatUNorm8Vec4srgb:
+        case HioFormatSNorm8:
+        case HioFormatSNorm8Vec2:
+        case HioFormatSNorm8Vec3:
+        case HioFormatSNorm8Vec4:
+            return 1;
+        case HioFormatUInt16:
+        case HioFormatUInt16Vec2:
+        case HioFormatUInt16Vec3:
+        case HioFormatUInt16Vec4:
+        case HioFormatInt16:
+        case HioFormatInt16Vec2:
+        case HioFormatInt16Vec3:
+        case HioFormatInt16Vec4:
+        case HioFormatFloat16:
+        case HioFormatFloat16Vec2:
+        case HioFormatFloat16Vec3:
+        case HioFormatFloat16Vec4:
+            return 2;
+        case HioFormatUInt32:
+        case HioFormatUInt32Vec2:
+        case HioFormatUInt32Vec3:
+        case HioFormatUInt32Vec4:
+        case HioFormatInt32:
+        case HioFormatInt32Vec2:
+        case HioFormatInt32Vec3:
+        case HioFormatInt32Vec4:
+        case HioFormatFloat32:
+        case HioFormatFloat32Vec2:
+        case HioFormatFloat32Vec3:
+        case HioFormatFloat32Vec4:
+            return 4;
+        default:
+            TF_CODING_ERROR("Unsupported type");
+            return 4;
+    }
+}
+
+/// Returns the number channels per pixel for the corresponding HioFormat
+static size_t
+_GetNumChannelsFromHIOFormat(HioFormat hioFormat)
+{
+    switch (hioFormat) {
+        case HioFormatUNorm8:
+        case HioFormatUNorm8srgb:
+        case HioFormatSNorm8:
+        case HioFormatUInt16:
+        case HioFormatInt16:
+        case HioFormatFloat16:
+        case HioFormatUInt32:
+        case HioFormatInt32:
+        case HioFormatFloat32:
+            return 1;
+        case HioFormatUNorm8Vec2:
+        case HioFormatUNorm8Vec2srgb:
+        case HioFormatSNorm8Vec2:
+        case HioFormatUInt16Vec2:
+        case HioFormatInt16Vec2:
+        case HioFormatFloat16Vec2:
+        case HioFormatUInt32Vec2:
+        case HioFormatInt32Vec2:
+        case HioFormatFloat32Vec2:
+            return 2;
+        case HioFormatUNorm8Vec3:
+        case HioFormatUNorm8Vec3srgb:
+        case HioFormatSNorm8Vec3:
+        case HioFormatUInt16Vec3:
+        case HioFormatInt16Vec3:
+        case HioFormatFloat16Vec3:
+        case HioFormatUInt32Vec3:
+        case HioFormatInt32Vec3:
+        case HioFormatFloat32Vec3:
+            return 3;
+        case HioFormatUNorm8Vec4:
+        case HioFormatUNorm8Vec4srgb:
+        case HioFormatSNorm8Vec4:
+        case HioFormatUInt16Vec4:
+        case HioFormatInt16Vec4:
+        case HioFormatFloat16Vec4:
+        case HioFormatUInt32Vec4:
+        case HioFormatInt32Vec4:
+        case HioFormatFloat32Vec4:
+            return 4;
+        default:
+            TF_CODING_ERROR("Unsupported type");
+            return 4;
+    }
+}
+
+// For compatability with Ice/Imr we transmogrify some matrix metadata
```

Comment:
```suggestion
// For compatability with legacy metadata keys we transmogrify some matrix metadata
```

Ice/Imr are Pixar-internal libraries, though these metadata keys also show up in OIIO and OpenEXR.  Anyway, hopefully makes things less confusing down the road.